webpack/localization-plugin-5/src/test/MixedAsyncDynamic.test.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+jest.disableAutomock();
+import { resolve } from 'path';
+import { promisify } from 'util';
+
+import webpack, { Compiler, Stats } from 'webpack';
+import { Volume } from 'memfs/lib/volume';
+
+import { LocalizationPlugin } from '../LocalizationPlugin';
+import type { ILocalizationPluginOptions } from '../interfaces';
+import { MemFSPlugin } from './MemFSPlugin';
+
+jest.setTimeout(1e9);
+
+async function testMixedAsyncDynamicInner(minimize: boolean): Promise<void> {
+  const memoryFileSystem: Volume = new Volume();
+  memoryFileSystem.fromJSON(
+    {
+      '/a/package.json': '{ "name": "a", "sideEffects": ["entry.js", "async.js", "asyncLoc.js"] }',
+      '/a/async.js': `console.log("blah");`,
+      '/a/asyncLoc.js': `import strings1 from './strings1.loc.json'; import strings2 from './strings2.loc.json'; console.log(strings1.test, strings2.another);`,
+      '/a/entry.js': `import(/* webpackChunkName: 'asyncLoc' */ './asyncLoc');import(/* webpackChunkName: 'async' */ './async');`,
+      '/a/strings1.loc.json': `{"test":{"value":"blah","comment":"A string"}}`,
+      '/a/strings2.loc.json': `{"another":{"value":"something else","comment":"Another string" }}`
+    },
+    '/'
+  );
+
+  const loader: string = resolve(__dirname, '../loaders/locjson-loader.ts');
+  const options: ILocalizationPluginOptions = {
+    localizedData: {
+      defaultLocale: {
+        localeName: 'en-us'
+      },
+      translatedStrings: {
+        foo: {
+          '/a/strings1.loc.json': {
+            test: 'baz'
+          },
+          '/a/strings2.loc.json': {
+            another: 'dead'
```

Comment:
None of the tests exercise missing translations. Can you add one that does?

We also don't test the passthrough locale or pseudolocales.